tests/unit_tests/scheduler/test_local_driver.py

```diff
@@ -0,0 +1,72 @@
+import asyncio
+
+import pytest
+
+from ert.scheduler.driver import JobEvent
+from ert.scheduler.local_driver import LocalDriver
+
+
+async def test_success(tmp_path):
+    driver = LocalDriver()
+
+    # LocalDriver doesn't have a polling task
+    assert driver.create_poll_task() is None
+
+    await driver.submit(42, "/usr/bin/env", "touch", "testfile", cwd=tmp_path)
+    assert await driver.event_queue.get() == (42, JobEvent.STARTED)
+    assert await driver.event_queue.get() == (42, JobEvent.COMPLETED)
+
+    assert (tmp_path / "testfile").exists()
+
+
+async def test_failure(tmp_path):
+    driver = LocalDriver()
+
+    await driver.submit(42, "/usr/bin/env", "false", cwd=tmp_path)
+    assert await driver.event_queue.get() == (42, JobEvent.STARTED)
+    assert await driver.event_queue.get() == (42, JobEvent.FAILED)
+
+
+async def test_file_not_found(tmp_path):
+    driver = LocalDriver()
+
+    await driver.submit(42, "/file/not/found", cwd=tmp_path)
+    assert await driver.event_queue.get() == (42, JobEvent.FAILED)
+
+
+async def test_kill(tmp_path):
+    driver = LocalDriver()
+
+    await driver.submit(42, "/usr/bin/env", "sleep", "10", cwd=tmp_path)
+    assert await driver.event_queue.get() == (42, JobEvent.STARTED)
+    await driver.kill(42)
+    assert await driver.event_queue.get() == (42, JobEvent.ABORTED)
+
+
+async def test_kill_unresponsive_process(tmp_path):
+    (tmp_path / "script").write_text(
+        """\
+    trap "" 1 2 3 4 5 6 7 8 10 11 12 13 14 15
+    sleep 10
+    """
+    )
+
+    driver = LocalDriver()
+
+    await driver.submit(42, "/bin/sh", tmp_path / "script", cwd=tmp_path)
+    assert await driver.event_queue.get() == (42, JobEvent.STARTED)
+    await driver.kill(42)
```

Comment:
Should we have some timout code here to assert that we are not at anytime ending waiting for 10 seconds for this test to pass?

Reply:
pytest.timeout for 9 seconds might suffice?

Reply:
Oh, I didn't notice you were referring to this specific test. Yes, I added `pytest.mark.timeout` and this test fails now, so it was indeed waiting the entire duration. :rofl:

Reply:
(that is strange, as I tested it and then it worked fine!)